src/org/mozilla/javascript/Parser.java

```diff
@@ -3272,7 +3272,10 @@ private AstNode arrayLiteral()
                     elements.add(new EmptyExpression(ts.tokenBeg, 1));
                     skipCount++;
                 }
-            } else if (tt == Token.RB) {
+            }else if(tt == Token.COMMENT) {
```

Comment:
If I'm going to suggest only one thing, I'd prefer if changes kept the same coding style (spaces, newlines, etc) as the code around it. Unfortunately applying any one particular style to this ancient codebase is a big job, but if we can at least keep things consistent with the lines around them then we'll all be happier I think.